### PR TITLE
Cmake: fix wrong cxx flags with single kokkos

### PR DESCRIPTION
**Pull request by RichardYCJ**
Fix the incorrect compile/link flags when built with single external kokkos, as mentioned in #14179

### CODE REVIEW

File: cmake/modules/FindKOKKOS.cmake [hunk 1]
```diff
@@ -80,8 +78,8 @@ ELSE()
     LIBRARIES REQUIRED KOKKOS_CORE_LIBRARY KOKKOS_CONTAINERS_LIBRARY
     INCLUDE_DIRS REQUIRED KOKKOS_INSTALL_INCLUDE_DIR
     USER_INCLUDE_DIRS REQUIRED KOKKOS_INSTALL_INCLUDE_DIR
-    CXX_FLAGS OPTIONAL KOKKOS_COMPILE_FLAGS
-    LINKER_FLAGS OPTIONAL KOKKOS_EXTRA_LD_FLAGS
+    CXX_FLAGS OPTIONAL ${KOKKOS_COMPILE_FLAGS}
+    LINKER_FLAGS OPTIONAL ${KOKKOS_EXTRA_LD_FLAGS}
```
Comment (tamiko) [inline cmake/modules/FindKOKKOS.cmake:82]: This needs to be without the enclosing `${...}` - provided both variables, `KOKKOS_COMPILE_FLAGS` and `KOKKOS_EXTRA_LD_FLAGS` contain the required flags.

Comment (RichardYCJ) [reply cmake/modules/FindKOKKOS.cmake:82]: >This needs to be without the enclosing ${...} - provided both variables

Hi @tamiko, I've also tried this one before. The turth is that the `KOKKOS_COMPILE(EXTRA_LD)_FLAGS ` only contains the **string** of the generator expression, and you need to use `${...}` to pass the actual expression which would transform to flags while generating the file cache and make file.

> But then I am not sure any more what the macro itself does.

Have no idea but that works at least :sweat_smile:

Comment (tamiko) [reply cmake/modules/FindKOKKOS.cmake:82]: The macro itself evaluates the string, i.e. looks at `${...}` - this is why I am surprised. I'll try to reproduce early next week.

Would you mind telling me a bit about your setup? (how do you get external kokkos)

**Review by masterleinad — CHANGES_REQUESTED** [review]
This doesn't work for me, see https://github.com/dealii/dealii/pull/14203. Can you verify that you actually get the correct flags and that they are not just being ignored with this pull request?

**Comment by masterleinad** [comment]
Superseded by #14203.

**Pull request closed without merging**